pandas/tests/test_groupby.py

```diff
@@ -2537,6 +2537,21 @@ def test_groupby_level_0_nonmulti(self):
         result = a.groupby(level=0).sum()
         self.assertEqual(result.index.name, a.index.name)
 
+    def test_groupby_list_and_range_levels_nonmulti(self):
+        # GH 13901
+        a = Series([1, 2, 3, 10, 4, 5, 20, 6], Index([1, 2, 3, 1,
+                                                      4, 5, 2, 6], name='foo'))
+
```

Comment:
Done. I took the liberty of modifying the test right above (which is the one I based mine off) to use `s = Series(...)` and `assert_series_equal`.